README.md

```diff
@@ -83,71 +86,125 @@ There are many common rules for naming and maintaining **Classes** within
     - `private`
 - Finally, sort alphabetically.
 
+Use [Pascal-case](#types-of-naming-conventions) for **Class Names**.
+
+- Always prefix interface names with an `I`
+- Always prefix abstract class names with an `A`
+
 Use **Namespaces** to ensure your scoping of items won't conflict with existing
 ones from other- or global namespaces. The project should at minimum use the
 project name for its namespaces in order to prevent conflicts with imported
 third party assets.
 
-All **Public Functions** should have a summary.
-```
+All **Public Functions** should have documentation in [the xmldoc format](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags),
+containing at least a summary and a description of their parameters and return
+value if any.
+```csharp
 /// <summary>
 /// Fire a gun
 /// </summary>
-public void Fire()
+/// <param name="bulletType">Type of the bullet to fire</param>
+/// <returns>True if the gun was able to fire, false if it wasn't for any reason</returns>
+public bool Fire(BulletType bulletType)
 {
-// Fire the gun.
+  // Fire the gun.
 }

```

Comment:
If that is too much, we can shorten the example, I mainly wanted to talk about the `xmldoc` format in addition to the `<summary>`  tag.

Reply:
I'm OK with this.